### PR TITLE
refactor(trading): migrate buy/sell quotes to useTradingQuoteRequest

### LINKED ISSUE #30213: Swap: "No offer found" flashes before the quote search actually runs
### Description
In the swap flow, after entering an amount the "No offer found" empty state is displayed immediately, before the quote request has actually run. The offer search only kicks in a few seconds later, and offers then populate normally.

### Steps to Reproduce
1. Go to Trade → Swap.
2. Select the send/receive assets and enter an amount.
3. Watch the offers area right after entering the amount.

### Expected Behavior
A loading/fetching state is shown while quotes are being retrieved. "No offer found" appears only after the search completes with zero results.

### Actual Behavior
"No offer found" shows instantly, then a few seconds later the search runs and offers appear — making it look like no offers exist when they do.

https://github.com/user-attachments/assets/d58f527b-23f2-4a09-bb1a-889c32037cfe

### PR DESCRIPTION
**Pull request by sherpaPSX**
## Description

Migrates buy & sell quotes onto the generic `useTradingQuoteRequest` hook (exchange already runs on it). Part of #26280 Phase 3. Also fixes a no-offer flash — folds `isScheduledQuotesRefresh` into `isFormLoading` so "no offer" no longer blinks during the debounce/fetch window.

## Notes for QA

Buy & sell: typing an amount must **not** flash "no offer" before quotes load; payment method still auto-selects. Sell: quotes clear when the send asset is deselected.

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/30213

<details>
<summary><h3>🤖 LLM Test Recommendations</h3></summary>

**Summary:** This change set modifies the core buy/sell quote-fetching hooks (useBuyQuotes, useSellQuotes) and their shared quote-request logic (useTradingQuoteRequest), along with the form initialization hooks (useTradingBuyForm, useTradingSellForm). These are central to all trading buy/sell flows, so the highest priority is given to E2E tests that directly exercise quote fetching, best-offer selection, and form validation across multiple assets (BTC, ETH, SOL, tokens). Navigation and dashboard asset tests are included at lower priority since they touch the same hooks but focus on other behaviors. The two unit test files changed have no E2E mapping since they are Jest/unit tests validating the hooks directly, and the useTradingQuoteRequest.ts file has no direct static mapping but is inferred to be exercised by all buy/sell quote tests since it's shared logic used by both useBuyQuotes and useSellQuotes.

<details><summary><b>Changed files (7)</b></summary>

- `packages/suite/src/hooks/wallet/trading/form/buy/__tests__/useBuyQuotes.test.tsx`
- `packages/suite/src/hooks/wallet/trading/form/buy/useBuyQuotes.ts`
- `packages/suite/src/hooks/wallet/trading/form/buy/useTradingBuyForm.ts`
- `packages/suite/src/hooks/wallet/trading/form/common/useTradingQuoteRequest.ts`
- `packages/suite/src/hooks/wallet/trading/form/sell/__tests__/useSellQuotes.test.tsx`
- `packages/suite/src/hooks/wallet/trading/form/sell/useSellQuotes.ts`
- `packages/suite/src/hooks/wallet/trading/form/sell/useTradingSellForm.ts`

</details>

**Recommended tests (9)**

<details><summary>🔴 <b>High priority (7)</b></summary>

- `suite/e2e/tests/trading/buy-bitcoin.test.ts` — Directly exercises the buy quotes flow (best offer calculation, provider selection, quote request/response handling) which is the core logic changed in useBuyQuotes.ts and useTradingQuoteRequest.ts.
- `suite/e2e/tests/trading/buy-ethereum.test.ts` — Covers the full buy quote-to-confirmation flow (best offer amount, quote sync) for a different asset, directly touching useBuyQuotes and useTradingBuyForm logic changed in this PR.
- `suite/e2e/tests/trading/buy-negative.test.ts` — Explicitly tests edge cases of the buy quote request (min/max limits, empty quotes, disabled best-offer button) which is precisely the logic being modified in useBuyQuotes.ts and useTradingQuoteRequest.ts.
- `suite/e2e/tests/trading/buy-solana-token.test.ts` — Exercises the buy quote sync and best offer display for an SPL token, directly validating the changed useBuyQuotes/useTradingBuyForm quote-fetching behavior across a different asset type.
- `suite/e2e/tests/trading/sell-ethereum-token.test.ts` — Directly exercises the sell quote flow (best offer, provider display) which maps to useSellQuotes.ts and useTradingSellForm.ts, both changed in this PR.
- `suite/e2e/tests/trading/sell-inputs.test.ts` — Validates sell form input handling, percentage buttons, and quote syncing across multiple assets (BTC, SOL), directly exercising useSellQuotes/useTradingSellForm and the shared quote request hook.
- `suite/e2e/tests/trading/sell-solana.test.ts` — Exercises the full sell quote-to-confirmation flow including offer comparison and best offer selection, directly touching the changed useSellQuotes/useTradingSellForm and useTradingQuoteRequest logic.

</details>

<details><summary>🟡 <b>Medium priority (1)</b></summary>

- `suite/e2e/tests/trading/navigation.test.ts` — Verifies that buy/sell/swap forms open correctly for various entry points and preselect the correct currency; this depends on useTradingBuyForm/useTradingSellForm initialization logic but doesn't exercise quote-fetching itself as deeply as other tests.

</details>

<details><summary>🟢 <b>Low priority (1)</b></summary>

- `suite/e2e/tests/dashboard/assets.test.ts` — Includes a buy button navigation test that touches useTradingBuyForm/useBuyQuotes indirectly, but the primary focus of this test is asset activation/discovery, not quote logic, so risk is lower.

</details>

⚠️ **Changes with no test coverage (2)**
- `packages/suite/src/hooks/wallet/trading/form/buy/__tests__/useBuyQuotes.test.tsx`
- `packages/suite/src/hooks/wallet/trading/form/sell/__tests__/useSellQuotes.test.tsx`

_Updated: 2026-07-28T12:48:33.775Z_

</details>

## 🌐 Preview deployments

🌐 **Suite Web preview:** [https://dev.suite.sldev.cz/suite-web/refactor/14-buysell-quotes-migration/web/](https://dev.suite.sldev.cz/suite-web/refactor/14-buysell-quotes-migration/web/)

## 🔍 Currents Test Results

🔍 **Suite desktop test results:** [View in Currents](https://app.currents.dev/projects/4ytF0E/runs?branch=refactor/14-buysell-quotes-migration&lookback=7)

🔍 **Suite web test results:** [View in Currents](https://app.currents.dev/projects/Og0NOQ/runs?branch=refactor/14-buysell-quotes-migration&lookback=7)

## 🔒 Quarantined E2E Tests

<details><summary>Trezor Suite (web) — 7 test(s)</summary>

| Test | Type |
|------|------|
| Recovery - dry run > Recovery with device reconnection | 🤖 auto |
| sol staking > display stake rewards on SOL staking account | 🤖 auto |
| All routes are covered by SECTIONS | 🤖 auto |
| stablecoin yield > User can deposit USDC into a yield vault | 🤖 auto |
| Quarantine test: "TrezorConnect webextension -> Suite Web,second call after popup was closed by user should work" | 🙋 manual |
| Quarantine test: "Recovery T2T1 - dry run,Recovery after partial recovery" | 🙋 manual |
| Firmware not ready | 🙋 manual |

</details>

_Updated: 2026-07-28T12:52:03.045Z • 7 test(s) total_

<details><summary>Trezor Suite (desktop) — 7 test(s)</summary>

| Test | Type |
|------|------|
| sol staking > display stake rewards on SOL staking account | 🤖 auto |
| Quarantine test: "Trading - Swap,Swap SOL to BTC" | 🙋 manual |
| Quarantine test: "Recovery - dry run,Recovery after partial recovery" | 🙋 manual |
| Quarantine test: "Recovery - dry run,Recovery with device reconnection" | 🙋 manual |
| Quarantine test: "Trading POC - passthru swap,Swap SOL to BTC with live quotes and blocked send" | 🙋 manual |
| Quarantine test: "Trading POC - passthru swap ETH,Swap ETH to BTC with live quotes and blocked send" | 🙋 manual |
| Firmware not ready | 🙋 manual |

</details>

_Updated: 2026-07-28T12:51:39.772Z • 7 test(s) total_

### CODE REVIEW

**Comment by sherpaPSX** [comment]
@coderabbit review

File: packages/suite/src/hooks/wallet/trading/form/buy/useBuyQuotes.ts [hunk 1]
```diff
@@ -23,140 +13,69 @@ import {
...
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [debouncedAmountKey, selectKey, isFetchAllowed]);
-
-    useTradingRefetchScheduler({
-        onRefetch: () => {
-            if (!isFetchAllowed) {
-                return;
-            }
-            fetchQuotes();
         },
```
[the remark this answers is not in the document]
Comment (sherpaPSX) [reply packages/suite/src/hooks/wallet/trading/form/buy/useBuyQuotes.ts:77]: Won't fix — the block is RHF-dependent, so it can't move to `@suite-common/trading`

File: packages/suite/src/hooks/wallet/trading/form/sell/__tests__/useSellQuotes.test.tsx [hunk 2]
```diff
@@ -201,11 +225,17 @@ describe('useSellQuotes', () => {
         await waitFor(() => expect(mockHandleRequest).toHaveBeenCalledTimes(1), { timeout: 1500 });
 
         await act(async () => {
-            result.current.setValue('outputs.0.currency', { value: 'eur', label: 'EUR' });
+            result.current.setValue('outputs.0.fiat', '99');
             await result.current.trigger();
         });
+        await wait(700);
+        expect(mockHandleRequest).toHaveBeenCalledTimes(1);
```
[the remark this answers is not in the document]
Comment (sherpaPSX) [reply packages/suite/src/hooks/wallet/trading/form/sell/__tests__/useSellQuotes.test.tsx:237]: Good point.

File: packages/suite/src/hooks/wallet/trading/form/sell/useSellQuotes.ts [hunk 3]
```diff
@@ -16,169 +8,91 @@ import {
...
+        isFetchAllowed: values => !!network && isSellQuotesFetchAllowed(values),
+        requestQuotes: values =>
+            dispatch(
+                sellThunks.handleRequestThunk({
+                    formValues: values,
+                    network: network!,
+                    shouldSendInSats,
+                    composeRequestCallback,
+                }),
+            ),
```
[the remark this answers is not in the document]
Comment (sherpaPSX) [reply packages/suite/src/hooks/wallet/trading/form/sell/useSellQuotes.ts]: Added guard to the useTradingQuoteRequest.ts

File: packages/suite/src/hooks/wallet/trading/form/sell/useTradingSellForm.ts [hunk 4]
```diff
@@ -127,17 +126,18 @@ export const useTradingSellForm = (): TradingSellFormContextProps => {
...
-        getValues,
-        setValue,
+    const { isScheduledQuotesRefresh } = useSellQuotes({
+        methods,
         network,
         shouldSendInSats,
         composeRequestCallback: () => {
             composeRequest(TRADING_FORM_OUTPUT_AMOUNT);
         },
     });
```
[the remark this answers is not in the document]
Comment (sherpaPSX) [reply packages/suite/src/hooks/wallet/trading/form/sell/useTradingSellForm.ts:136]: This can't loop: composeRequestCallback only fires when `setMaxOutputId === undefined`

File: packages/suite/src/hooks/wallet/trading/form/buy/useTradingBuyForm.ts [hunk 5]
```diff
@@ -105,12 +106,14 @@ export const useTradingBuyForm = (): TradingBuyFormContextProps => {
...
     const formIsValid = Object.keys(formState.errors).length === 0;
     const hasValues = (fiatInput || cryptoInput) && !!currencySelect?.value;
-    const isFormLoading = formState.isSubmitting || isLoading;
+    const isFormLoadingBase = formState.isSubmitting || isLoading;
     const isFormInvalid = !(formIsValid && hasValues) || !isReceiveAddressFormValid;
-    const isLoadingOrInvalid = noProviders || isFormLoading || isFormInvalid;
 
     // based on selected cryptoSymbol, because of using for validation cryptoInput
     const network = getNetwork(cryptoSelect?.networkSymbol ?? TRADING_DEFAULT_CRYPTO_CURRENCY);
+    const { isBtcSatsAmountUnit: shouldSendInSats } = useBitcoinAmountUnit(
```
Comment (vytick) [inline packages/suite/src/hooks/wallet/trading/form/buy/useTradingBuyForm.ts:114]: __thought:__ this can probably be part of the `useBuyQuotes` as well. Wdyt?

Comment (sherpaPSX) [reply packages/suite/src/hooks/wallet/trading/form/buy/useTradingBuyForm.ts:114]: I will do it as a follow-up https://github.com/trezor/trezor-suite/issues/30479 in all scopes

**Review by vytick — APPROVED** [review]
